lib/semaphore.rb

```diff
@@ -5,20 +5,97 @@
 # It is based on Redis::Semaphore and can be extended to allow multiple
 # processes to enter the critical path.
 # Currently, only a mutex is implemented in this abstraction.
+#
+# Expiration is given in seconds. Can be nil. If set, the Semaphore will unlock
+# after `expiration` seconds.
+#
+# It takes a block containing the critical path.
 class Semaphore
-  def self.exclusively(lock_key, &block)
-    if defined?(Sidekiq::Testing) && Sidekiq::Testing.inline?
-      yield
-    else
+  SEMAPHORES_NAMESPACE =
+    "#{Settings.redis.namespace}:#{self.to_s.downcase}".freeze
```

Comment:
I'd rather leave this one in there to make sure one understands that this is the class name